xc7/primitives/slicem/slicem.pb_type.xml

```diff
@@ -299,36 +299,34 @@
         <direct name="B_WA8" input="SLICEM_MODES.WA8"  output="B_DRAM.WA[7]" />
         <direct name="A_WA8" input="SLICEM_MODES.WA8"  output="A_DRAM.WA[7]" />
 
-        <!-- Direct DI1 inputs -->
-        <direct name="DI" input="SLICEM_MODES.DI" output="D_DRAM.DI1" />
-        <direct name="CI" input="SLICEM_MODES.CI" output="C_DRAM.CI" />
-        <direct name="BI" input="SLICEM_MODES.BI" output="B_DRAM.BI" />
-        <direct name="AI" input="SLICEM_MODES.AI" output="A_DRAM.AI" />
-
-        <!-- Parent DI1 inputs -->
-        <direct name="P_CI" input="SLICEM_MODES.DI" output="C_DRAM.PARENT_DI" />
-        <direct name="P_BI" input="SLICEM_MODES.DI" output="B_DRAM.PARENT_DI" />
-
-        <!-- This isn't quiet right.  SLICEM_MODES.DI is used if and only if
-             B_DRAM is in dual-port mode, and SLICEM_MODES.BI is used if and
-             only if B_DRAM is in single-port mode.
-
-             Synthesis should only emit A_DRAM.PARENT_DI connection that match
-             with the B_DRAM mode.
-
-             The reason this isn't modelled in VPR is that VPR cannot handle
-             mode based pure edges (e.g. input -> output without touching a
-             block). If VPR is modified to handle mode-based pure connections,
-             then this can be fully modelled in a mode based connection,
-             rather than a simple mux.
-
-             It is worth noting that if B_DRAM only have 1 single port mode,
-             this issue wouldn't arise, but that would require creating
-             B_DRAM32 and another top level model for 32-bit DRAMs,
-             which would duplicate SLICEM_MODES more than would otherwise be
-             required.
-             -->
-        <mux name="P_AI" input="SLICEM_MODES.DI SLICEM_MODES.BI" output="A_DRAM.PARENT_DI" />
+        <!-- DI1 inputs -->
+        <direct name="D_DI1" input="SLICEM_MODES.DI" output="D_DRAM.DI1" />
+        <mux name="CDI1MUX" input="SLICEM_MODES.DI SLICEM_MODES.CI" output="C_DRAM.DI1">
+          <metadata>
+            <meta name="fasm_mux">
+              SLICEM_MODES.DI = NULL
+              SLICEM_MODES.CI = DI1MUX.CI
```

Comment:
This should probably be CLUT.DI1MUX.CI?

Reply:
The prefix includes `CLUT` already.

Reply:
Does it?  This is one level higher than before, this is the SLICEM level?

Reply:
Yes, I checked the FASM when I had an error due to calling it `CDI1MUX.CI` which matches the documentation. I looked at the database and changed it to `CLUT.DI1MUX.CI`, which resulted in `...CLUT.CLUT.DI1MUX.CI`, which of course failed.

So now it's just `DI1MUX.CI`, which results in `...CLUT.DI1MUX.CI`, which works.

Reply:
I'm not sure where it gets `CLUT` from, though. I just checked that it worked.

Reply:
Probably worth fixing, probably a FASM bug.